.projenrc.ts

```diff
@@ -19,20 +21,25 @@ const project = new awscdk.AwsCdkConstructLibrary({
     module: "cdk_circleci_oidc",
   },
 
+  // deps: [],
+  devDeps: ["@mrgrain/jsii-struct-builder"],
   depsUpgrade: false,
 
+  eslintOptions: {
+    dirs: ["src"],
+    ignorePatterns: ["src/generated/*.ts"], // ignore generated files
+  },
   prettier: true,
   prettierOptions: {
     settings: {
       printWidth: 120,
       proseWrap: ProseWrap.ALWAYS,
     },
   },
-
-  // deps: [],                /* Runtime dependencies of this module. */
-  // description: undefined,  /* The description is just a string that helps people understand the purpose of the package. */
-  // devDeps: [],             /* Build dependencies for this module. */
-  // packageName: undefined,  /* The "name" in package.json. */
 });
 
+new ProjenStruct(project, { name: "RoleProps", filePath: "src/generated/IamRoleProps.ts" }).mixin(
+  Struct.fromFqn("aws-cdk-lib.aws_iam.RoleProps").omit("assumedBy").withoutDeprecated(),
+);
```

Comment:
JSII can't handle statements like:

```ts
type MyRoleProps = Omit<RoleProps, 'assumedBy'>;
```

So I'm using jsii-struct-builder to accomplish the same thing.